src/main/java/com/qunite/api/service/QueueServiceImpl.java

```diff
@@ -51,12 +54,47 @@ public Optional<Integer> getMemberPositionInQueue(Long memberId, Long queueId) {
         .map(entry -> entry.getEntryIndex() + 1);
   }
 
+  @Override
+  @Transactional(isolation = Isolation.REPEATABLE_READ)
+  public void changeMemberPositionInQueue(Long memberId, Long queueId,
+                                          Integer newIndex, String username) {
+    var entry = entryRepository.findById(new EntryId(memberId, queueId))
+        .orElseThrow(() -> new EntryNotFoundException(
+            "Could not find entry by memberId %s and queueId %s".formatted(memberId, queueId)));
+    if (isUserQueueCreatorOrManagerByCredentials(queueId, username)) {
+      var currentIndex = entry.getEntryIndex();
+      if (!currentIndex.equals(newIndex)) {
+        int startIndex = Math.min(currentIndex, newIndex);
+        int endIndex = Math.max(currentIndex, newIndex);
+        int increment = currentIndex < newIndex ? -1 : 1;
+
+        entryRepository.updateEntryIndexes(queueId, startIndex, endIndex, increment);
+        entry.setEntryIndex(newIndex);
+      }
+    } else {
+      throw new AccessDeniedException("User is not a creator or manager");
+    }
+  }
+
+  @Override
+  @Transactional(isolation = Isolation.REPEATABLE_READ)
+  public void deleteMemberFromQueue(Long memberId, Long queueId, String username) {
+    var entry = entryRepository.findById(new EntryId(memberId, queueId))
+        .orElseThrow(() -> new EntryNotFoundException(
+            "Could not find entry by memberId %s and queueId %s".formatted(memberId, queueId)));
+    if (isUserQueueCreatorOrManagerByCredentials(queueId, username)) {
+      entryRepository.deleteById(entry.getId());
+      entryRepository.decrementEntryIndexes(entry.getQueue().getId(), entry.getEntryIndex());
+    } else {
+      throw new AccessDeniedException("User is not a creator or manager");
```

Comment:
Perhaps it is better to prefer to throw out always our mistakes, then we will be sure that it is our mistake for sure